### PR TITLE
criocli: fix/diff-friendlier `zsh` completion generation.

### PR DESCRIPTION
**Pull request by klihub**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Commands with a single quote (`'`) in their usage string cause syntactically incorrect `zsh` completion generation. This patchset aims to fix that, while it also reformats the generated `zsh` completions for easier diffing with less contextual noise in the future.

#### Which issue(s) this PR fixes:

None

#### Does this PR introduce a user-facing change?

```release-note
Fix zsh completion generation.
```

### CODE REVIEW

**Comment by haircommander** [comment]
/ok-to-test
/approve

LGTM, thanks @klihub

**Comment by haircommander** [comment]
@cri-o/cri-o-maintainers PTAL

**Review by fidencio — COMMENTED** [review]
@klihub, just a small suggestion, which could be applied to pretty much all blocks changed.
Please, let me know what do you think.

File: completions/zsh/_crio
Comment (TomSweeneyRedHat) [reply completions/zsh/_crio:15]: The same below too @fidencio ?

**Comment by TomSweeneyRedHat** [comment]
LGTM once tests are happy and @fidencitio suggestions are resolved.

File: completions/zsh/_crio
Comment (klihub) [reply completions/zsh/_crio]: Makes sense. I updated the implementation accordingly.

File: completions/zsh/_crio
Comment (klihub) [reply completions/zsh/_crio]: Updated the implementation accordingly.

**Comment by fidencio** [comment]
LGTM, thanks @klihub!

**Comment by haircommander** [comment]
/retest-required

(if it continues to fail we can override e2e-agnositc, gcp passed)

ping @saschagrunert for final review/tag?

**Review by saschagrunert — APPROVED** [review]
/retest-required
/lgtm

**Pull request merged**
